plugins/modules/ipadnsconfig.py

```diff
@@ -0,0 +1,261 @@
+            if not (is_ipv4_addr(ip_address) or is_ipv6_addr(ip_address)):
+                module.fail(
+                    msg="Invalid IP for DNS forwarder: %s" % ip_address)
+            if port is None:
+                _forwarders.append(ip_address)
+            else:
+                _forwarders.append('%s port %d' % (ip_address, port))
+
+        global_forwarders = dnsconfig.get('idnsforwarders', [])
+        if state == 'absent':
+            _args['idnsforwarders'] = [
+                fwd for fwd in global_forwarders if fwd not in _forwarders]
+            # When all forwarders should be excluded, use an empty string ('').
+            if not _args['idnsforwarders']:
+                _args['idnsforwarders'] = ['']
+
+        elif state == 'present':
+            _args['idnsforwarders'] = [
+                fwd for fwd in _forwarders if fwd not in global_forwarders]
+            # If no forwarders should be added, remove argument.
+            if not _args['idnsforwarders']:
+                del _args['idnsforwarders']
+
+        else:
+            # shouldn't happen, but let's be paranoid.
+            module.fail(msg="Invalid state: %s" % state)
+
+    if forward_policy is not None:
+        _args['idnsforwardpolicy'] = forward_policy
+
+    if allow_sync_ptr is not None:
+        _args['idnsallowsyncptr'] = 'TRUE' if allow_sync_ptr else 'FALSE'
+
+    return _args
+
+
+def check_parameters(module):
+    pass
+
+
+def main():
+    forwarder_spec = dict(
+       ip_address=dict(type=str, required=True),
+       port=dict(type=int, required=False, default=None)
+    )
+
+    ansible_module = AnsibleModule(
+       argument_spec=dict(
+           # general
+           ipaadmin_principal=dict(type='str', default='admin'),
+           ipaadmin_password=dict(type='str', no_log=True),
+
+           # dnsconfig
+           forwarders=dict(type='list', default=None, required=False,
+                           options=dict(**forwarder_spec)),
+           forward_policy=dict(type='str', required=False, default=None,
+                               choices=['only', 'first', 'none']),
+           allow_sync_ptr=dict(type='bool', required=False, default=None),
+
+           # general
+           state=dict(type="str", default="present",
+                      choices=["present", "absent"]),
+
+       )
+    )
+
+    ansible_module._ansible_debug = True
+
+    # general
+    ipaadmin_principal = module_params_get(ansible_module,
+                                           "ipaadmin_principal")
+    ipaadmin_password = module_params_get(ansible_module,
+                                          "ipaadmin_password")
+
+    forwarders = module_params_get(ansible_module, 'forwarders') or []
+    forward_policy = module_params_get(ansible_module, 'forward_policy')
+    allow_sync_ptr = module_params_get(ansible_module, 'allow_sync_ptr')
+
+    state = module_params_get(ansible_module, 'state')
+
+    # Check parameters.
```

Comment:
Shouldn't this checks be inside the `check_parameters` function. If not do we really need to define the empty function?

Reply:
The empty function is unnecessary. Removed it.